sharktank/sharktank/layers/ffn_block.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+
+from .base import Theta, ThetaLayer
+from .linear import LinearLayer
+
+__all__ = [
+    "FFN",
+]
+
+
+class FFN(ThetaLayer):
+    def __init__(
+        self,
+        theta: Theta,
+        expert_idx: Optional[int] = None,
+    ):
+        super().__init__(theta)
+        ffn_g = "ffn_gate"
+        ffn_u = "ffn_up"
+        ffn_d = "ffn_down"
+        if expert_idx is not None:
```

Comment:
We can remove the MOE aspect from this FFN, as we will only be using `ffn_moe_block.py` for MOE models. Ideally we can combine both, but for now `ffn_moe_block.py` has all the logic to handle an MOE model.

Reply:
I think its ok to keep separate, but I'll remove the expert index logic from this one